model/riscv_ext_regs.sail

```diff
@@ -1,25 +1,22 @@
-/*=======================================================================================*/
-/*  This Sail RISC-V architecture model, comprising all files and                        */
-/*  directories except where otherwise noted is subject the BSD                          */
-/*  two-clause license in the LICENSE file.                                              */
-/*                                                                                       */
-/*  SPDX-License-Identifier: BSD-2-Clause                                                */
-/*=======================================================================================*/
+// =======================================================================================
+// This Sail RISC-V architecture model, comprising all files and
+// directories except where otherwise noted is subject the BSD
+// two-clause license in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+// =======================================================================================
 
-/* This file contains register handling functions that can be
- * overridden by extensions.
- */
+// This file contains register handling functions that can be
+// overridden by extensions.
 
-/*!
-THIS(csrno, priv, isWrite) allows an extension to block access to csrno,
-at Privilege level priv. It should return true if the access is allowed.
-*/
```

Comment:
Not sure what the equivalent line comment version of doc comments is in sail?

In LLVM function doc comments generally use a triple slash `///` (which some IDEs understand and highlight them differently and parse the markup inside the doc comment.

The other style I've seen is `//!` which is apparently also valid for doxygen https://www.doxygen.nl/manual/docblocks.html.

I don't think we are currently using any of the doc comments here, but in the past we were using the function annotations in the sail model to generate nicely formatted LaTeX for the CHERI specification.

I think the /*! comments should use whatever line-doc-comment style sail understands (@Alasdair?).

Reply:
Ah is that something Sail understands? I thought it was just the author being weird :-D

Reply:
Yes it just substitutes the name of the identifier you are documenting so you don't have to re-write it every time you want to reference it.